### PR TITLE
NIFI-11513 Remove deprecated NarProvider and implementations

### PR DESCRIPTION
**Pull request by exceptionfactory**
# Summary

[NIFI-11513](https://issues.apache.org/jira/browse/NIFI-11513) Removes the deprecated `NarProvider` interface from `nifi-api` and also removes the `HDFSNarProvider` and `NiFiRegistryNarProvider` implementations.

Additional changes include other framework module updates, including removing the `nifi-external-resource-utils` module from `nifi-commons`, removing the deprecated `SimpleSwapSerializer`, and removing deprecation annotations from methods in `EncryptedSchemaRepositoryRecordSerde` which implement non-deprecated interface methods.

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [X] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [X] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [X] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [X] Pull Request based on current revision of the `main` branch
- [X] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [X] Build completed using `mvn clean install -P contrib-check`
  - [X] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/src/main/resources/conf/nifi.properties [hunk 1]
```diff
@@ -383,7 +383,3 @@ nifi.performance.tracking.percentage=${nifi.performance.tracking.percentage}
 #   nifi.nar.library.provider.hdfs.kerberos.principal=nifi@NIFI.COM
 #   nifi.nar.library.provider.hdfs.kerberos.keytab=/path/to/nifi.keytab
 #   nifi.nar.library.provider.hdfs.kerberos.password=
-#
-# Example NiFi Registry NAR Provider:
-#   nifi.nar.library.provider.nifi-registry.implementation=org.apache.nifi.registry.extension.NiFiRegistryNarProvider
-#   nifi.nar.library.provider.nifi-registry.url=http://localhost:18080
```
Comment (bbende) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/src/main/resources/conf/nifi.properties:389]: Could we leave this block in the properties file, but just fix the class name from `NiFiRegistryNarProvider` to `NiFiRegistryExternalResourceProvider` ?

Comment (exceptionfactory) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/src/main/resources/conf/nifi.properties:389]: Good point, will make the adjustment.

Comment (exceptionfactory) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/src/main/resources/conf/nifi.properties:389]: @bbende Pushed an update restoring the Registry example using the preferred Provider class name.

**Review by bbende — APPROVED** [review]
Looks good, going to merge

**Pull request merged**